### PR TITLE
feat: Sensible non-Error exception serializer

### PR DESCRIPTION
**Pull request by kamilogorek**
Initial attempt to tackle  https://github.com/getsentry/raven-node/issues/316 and https://github.com/getsentry/raven-node/issues/187

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -355,20 +356,36 @@ extend(Raven.prototype, {
...
       cb = kwargs;
       kwargs = {};
     } else {
       kwargs = kwargs || {};
     }
 
+    if (!(err instanceof Error)) {
+      if (utils.isPlainObject(err)) {
+        // This will allow us to group events based on top-level keys
+        // which is much better than creating new group when any key/value change
```
Comment (benvinegar) [inline lib/client.js]: Damn, that's a great idea.

File: lib/client.js [hunk 2]
```diff
@@ -355,20 +356,36 @@ extend(Raven.prototype, {
...
+        var hash = md5(Object.keys(err).sort());
+        var message = 'Sentry: non-Error exception captured [' + hash + ']';
```
Comment (benvinegar) [inline lib/client.js]: I don't know that we need to prefix `Sentry:` here.

What about:

```
Non-Error exception captured with keys foo, bar, ...
```

^Something like that, only showing the first 2 or 3 keys.

Comment (kamilogorek) [reply lib/client.js]: Updated

File: lib/client.js (hunk 1 above)
Comment (kamilogorek) [reply lib/client.js]: image

Comment (MaxBittker) [reply lib/client.js]: ah, that's awesome kamil. do they need to be hashed? I guess it's a length thing, right?

Comment (kamilogorek) [reply lib/client.js]: Without hashing, very large objects will create way too large fingerprints. This way it's always fixed length as you mentioned.

**Review by HazAT — APPROVED** [review]

**Pull request merged**